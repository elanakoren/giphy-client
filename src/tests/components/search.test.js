//@flow
import React from 'react';
import Search from '../../components/search';
import {mount, shallow} from 'enzyme';

describe('search', () => {
    let wrapper, fetchGifs;
    beforeEach(() => {
        fetchGifs = jest.fn();
        wrapper = mount(<Search {...{
            gifActions: {fetchGifs},
        }}/>);
    });

    describe('clicking the search button', () => {
        it('calls the fetchGifs function', () => {
            wrapper.find('.submit-button').simulate('click');
            expect(fetchGifs).toHaveBeenCalledWith(expect.any(HTMLInputElement));
        });
    });

    describe('clicking the clear button', () => {
       beforeEach(() => {
           wrapper.find('input').simulate('keydown', { which: 'a' });
       });

       it('clears the input and fetches a new batch of trending gifs', () => {
           wrapper.find('.clear-button').simulate('click');
           expect(fetchGifs).toHaveBeenCalled();
           expect(fetchGifs).not.toHaveBeenCalledWith(expect.any(HTMLInputElement));
           expect(wrapper.find('input').props().value).toEqual(undefined)
       });
    });

    describe('the snapshot', () => {
        beforeEach(() => {
            fetchGifs = jest.fn();
            wrapper = shallow(<Search {...{
                gifActions: {fetchGifs},
            }}/>);
        });

        it('matches the snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });
    });
});