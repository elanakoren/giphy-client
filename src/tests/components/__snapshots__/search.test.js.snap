// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search the snapshot matches the snapshot 1`] = `
<div
  className="search-control"
>
  <input
    type="text"
  />
  <button
    className="search-button submit-button"
    onClick={[Function]}
  >
    Search
  </button>
  <button
    className="search-button clear-button"
    onClick={[Function]}
  >
    Clear Search
  </button>
</div>
`;
